superset/assets/src/SqlLab/reducers.js

```diff
@@ -157,7 +157,7 @@ export const sqlLabReducer = function (state = {}, action) {
         progress: 100,
         results: action.results,
         rows,
-        state: action.query.state,
+        state: 'rendering',
```

Comment:
Is this where the code was setting state to `success` earlier?

Reply:
Nope. The `state` is still `running` and that's why the inconsistency between progress bar and status happens. 

So after query succeeds, the `action.results` is returned from the server containing something like
```
{
  'query': {
    'id': ...,
    'state': 'success',
    'progress': 100,
    ... 
  }
  'status': 'success'
}```
but `action.query` itself is a part of the state of the app which contains `{'state': 'running'}` field in itself.

Then in the next round of rerendering, the query data returned from the server will show in the sqllab `SouthPane` and `state` will be changed in `QueryAutoRefresh`: https://github.com/apache/incubator-superset/blob/cb92739c46355c3217b33297cbcce8801f5768aa/superset/assets/src/SqlLab/components/QueryAutoRefresh.jsx#L45

Please correct me if my understanding is wrong.

Reply:
Got it. Thanks for explaining.